### DISCUSSION TITLE
[HttpClient] Is there an HttpClientInterface adapter for KernelBrowser

### DISCUSSION THREAD

**Discussion by rmikalkenas** [symfony/symfony#47023] [Q&A] [answered]
The title says it all.
Use case: I have a library which implements an api client for my application. I would like to write an automatic test which would call my api throughout symfony's http client, but I dont want to execute "actual" http request, instead I would like to reuse KernelBrowser for it.
Simple example:
```
$apiClient = new ApiClient(\Symfony\Component\HttpClient\HttpClient::create());
$data = $apiClient->getData(); // does actual http request and returns readable data object
```
in the test I would like to have something similar:
```
// WebTestCase
$apiClient = new ApiClient(new KernelBrowserHttpClient(self::createClient()));
$data = $apiClient->getData(); // does actual http request and returns readable data object
```

**Comment by stof** [comment, marked as answer]
There's no such thing

**Reply by rmikalkenas** [reply]
I see..
Looks like it would be a very good improvement! I will try to work on it and contribute to symfony's http client
